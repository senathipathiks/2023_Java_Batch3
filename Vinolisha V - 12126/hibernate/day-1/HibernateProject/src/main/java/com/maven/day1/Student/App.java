package com.maven.day1.Student;

import org.hibernate.*;
import org.hibernate.cfg.Configuration;
public class App
{
    public static void main( String[] args )
    {
    	Student s1 = new Student(1,"vinolisha","Nagercoil");
    	Student s2=new Student(2, "Vino" , "Chennai");
    	Student s3=new Student(3,"Balu","Tirutani");
    	
    	
    	Configuration cfg = new Configuration().configure("hibernate.cfg.xml");
    	SessionFactory factory=cfg.buildSessionFactory(); //will connect the connection database
    	Session session = factory.openSession();
    	Transaction t = session.beginTransaction(); //provide the commit and rollback
    	session.save(s1);
    	session.save(s2);
    	session.save(s3);
    	t.commit();
    }
}