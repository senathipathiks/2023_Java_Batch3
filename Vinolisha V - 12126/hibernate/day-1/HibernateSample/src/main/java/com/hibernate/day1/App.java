package com.hibernate.day1;

import java.util.List;
import java.util.Scanner;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;



public class App 
{
    public static void main( String[] args )
    {
    	
    	Scanner sc=new Scanner(System.in);
    	
    	Course c=new Course();
    	
    	Configuration cfg=new Configuration().configure().addAnnotatedClass(Course.class);
    	SessionFactory factory=cfg.buildSessionFactory();
    	Session session=factory.openSession();
    	Transaction t=session.beginTransaction();
    	
    	boolean task=true;
    	
    	while(task) {
    	
    	System.out.println("Welcome to Course Page");
    	System.out.println("1.Add \n2.Delete \n3.U-p-date \n4.View \n5.ViewAll \5Exit Applicatio*-");
    	System.out.println("Select your choice");
    	int choice=sc.nextInt();
    	
    	switch (choice) {
		case 1:
			System.out.println("Enter the Id");
			int id=sc.nextInt();
			System.out.println("Enter the name");
			String name=sc.next();
			System.out.println("Enter the Address");
			String addr=sc.next();
			System.out.println("Enter the Salary");
			float sal=sc.nextFloat();
			c.setId(id);
			c.setName(name);
			c.setAddress(addr);
			c.setSalary(sal);
			session.save(c);
			t.commit();
			
			break;
		
		case 2:
			System.out.println("Enter the Id");
			int id1=sc.nextInt();
			c =session.find(Course.class, id1);
			session.delete(c);
			t.commit();
			
			break;
			
		case 3:
			System.out.println("Enter the Id");
			int id2=sc.nextInt();
			c=session.find(Course.class, id2);
			
			System.out.println("Enter the name");
			String name1=sc.next();
			System.out.println("Enter the Address");
			String addr1=sc.next();
			System.out.println("Enter the Salary");
			float sal1=sc.nextFloat();
			
			c.setName(name1);
			c.setAddress(addr1);
			c.setSalary(sal1);
			session.save(c);
			t.commit();
			
			break;
			
		case 4: //view details
			System.out.println("Enter the id want to view");
			int id3=sc.nextInt();
			c=session.find(Course.class, id3);
			System.out.println(c);
			
			break;
			
			
		case 5: //viewall details
	    	List<Course> courseList=session.createQuery("from Course").list();
	    	for(Course c1 : courseList) {
				System.out.println(c1);
			}
	    	
	    	break;
	    	
		case 6:
			System.out.println("Thanks for using the application");
			System.exit(0);
			task=false;

		default:
			System.out.println("Select the correct choice");
			break;
		}
    	
    	
    		
    	
    	
    	}
    	
    }
}
