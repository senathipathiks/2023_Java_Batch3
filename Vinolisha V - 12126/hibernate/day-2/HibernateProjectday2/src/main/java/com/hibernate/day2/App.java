package com.hibernate.day2;

import java.util.List;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

public class App 
{
    public static void main( String[] args )
    {
    	
//    	StaffPojoClass staff= new StaffPojoClass(1,"vino","Attoor");
//    	StaffPojoClass staff1= new StaffPojoClass(2,"vinolisha","Chennai");
//    	StaffPojoClass staff2= new StaffPojoClass(3,"vinisha","Coimbatore");
//    	StaffPojoClass staff3= new StaffPojoClass(4,"Prabha","Nagercoil");
//    	StaffPojoClass staff4= new StaffPojoClass(5,"Priya","Punjab");
    	
    	Configuration cfg=new Configuration().configure();
    	SessionFactory factory=cfg.buildSessionFactory();
    	Session session=factory.openSession();
    	Transaction t=session.beginTransaction();
    	
//    	session.save(staff);
//    	session.save(staff1);
//    	session.save(staff2);
//    	session.save(staff3);
//    	session.save(staff4);
    	//t.commit();
    	
    	//update
//    	StaffPojoClass temp=session.find(StaffPojoClass.class, 4);
//    	temp.setName("Sherin");    	
//    	session.update(temp);
    	//t.commit();
    	
    	
    	//delete
//    	StaffPojoClass temp=session.find(StaffPojoClass.class, 4);
//    	session.delete(temp);
//    	
//    	t.commit();
    	
    	//view details
    	List<StaffPojoClass> staffList=session.createQuery("from StaffPojoClass").list();
    	for(StaffPojoClass s : staffList) {
			System.out.println(s);
		}
    	
    	System.out.println("Done");
    	
    	
    }
}
