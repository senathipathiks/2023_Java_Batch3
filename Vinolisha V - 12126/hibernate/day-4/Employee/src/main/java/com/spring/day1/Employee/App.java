package com.spring.day1.Employee;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;



/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Laptop l1=new Laptop(142,"Lenova");
        Laptop l2=new Laptop(155, "Dell");
        
        session.persist(l1);
        session.persist(l2);
        
        Employee e1=new Employee(106,"Viisha",l1);
        Employee e2=new Employee(107,"Vinolisha",l2);
        
        session.persist(e1);
        session.persist(e2);
        
        t.commit();
        
    }
}
