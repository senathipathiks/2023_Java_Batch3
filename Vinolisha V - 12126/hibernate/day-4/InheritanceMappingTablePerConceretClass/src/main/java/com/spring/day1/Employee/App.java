package com.spring.day1.Employee;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;



/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
       Employee e1=new RegularEmp(101,"Sena",10000.00f,1000);
       Employee e2=new ContractEmp(102,"Gowtham",10000.00f,"2 years");
       
       session.save(e1);
       session.save(e2);
       
       t.commit();
       System.out.println("Success");
       
    }
}
