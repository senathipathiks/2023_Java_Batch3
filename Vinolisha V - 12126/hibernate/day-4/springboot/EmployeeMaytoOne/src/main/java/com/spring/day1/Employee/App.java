package com.spring.day1.Employee;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;



/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Department d1=new Department(143,"Sales");
        Department d2=new Department(144,"Marketing");
        Department d3=new Department(145,"Marketing");
        Employee e1=new Employee(108,"Sherin",d1);
        Employee e2=new Employee(109,"Brainy",d1);
        Employee e3=new Employee(110,"Fathi",d2);
        Employee e4=new Employee(111,"Priya",d3);
        
        session.persist(d1);
        session.persist(d2);
        session.persist(d3);
        
       
        
        session.persist(e1);
        session.persist(e2);
        session.persist(e3);
        session.persist(e4);
        
        t.commit();
        
    }
}
