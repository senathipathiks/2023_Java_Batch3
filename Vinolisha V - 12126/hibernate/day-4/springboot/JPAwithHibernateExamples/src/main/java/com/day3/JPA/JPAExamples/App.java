package com.day3.JPA.JPAExamples;



import jakarta.persistence.EntityManager;
import jakarta.persistence.EntityManagerFactory;
import jakarta.persistence.EntityTransaction;
import jakarta.persistence.Persistence;

public class App {
	public static void main(String[] args) {
		
		EntityManagerFactory emf = Persistence.createEntityManagerFactory("course_db");

		EntityManager em = emf.createEntityManager();
		
		System.out.println(emf);
		EntityTransaction t = em.getTransaction();
		Course c1 = new Course(103, "C++", "CSE");
		System.out.println(t);
		
		t.begin();
		em.persist(c1);
		t.commit();
		
		emf.close();

	}
}
