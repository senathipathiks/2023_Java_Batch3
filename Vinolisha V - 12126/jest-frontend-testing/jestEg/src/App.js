import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Welcome Vinolisha</h1>
      <br/>
      <h2>Welcome to It Company</h2>
      <br/>
      <h2>Welcome Jothi Kirubs</h2>
    </div>
  );
}

export default App;
