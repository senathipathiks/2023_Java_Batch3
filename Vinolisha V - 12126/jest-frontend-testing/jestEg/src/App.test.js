import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from 'react-test-renderer';

// describe("Test Suite for App", () => {
  // test("renders Welcome Vinolisha", () => {
  //   render(<App />);
  //   const linkElement = screen.getByText("Welcome Vinolisha");
  //   expect(linkElement).toBeInTheDocument();
  // });

  // it("renders Welcome to It Company", () => {
  //   render(<App />);
  //   const linkElement = screen.getByText("Welcome to It Company");
  //   expect(linkElement).toBeInTheDocument();
  // });
// });
 
  test('Snapshot Testing', () => {
    const comp = renderer.create(<App />).toJSON();
    expect(comp).toMatchSnapshot();
  });

