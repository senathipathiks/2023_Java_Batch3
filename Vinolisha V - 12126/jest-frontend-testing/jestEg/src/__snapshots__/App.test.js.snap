// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing 1`] = `
<div
  className="App"
>
  <h1>
    Welcome Vinolisha
  </h1>
  <br />
  <h2>
    Welcome to It Company
  </h2>
  <br />
  <h2>
    Welcome Jothi Kirubs
  </h2>
</div>
`
;
